### PR TITLE
Bugfix plugin 1033

### PR DESCRIPTION
**Pull request by kunalsharma-repo**

### CODE REVIEW

**Review by rmstar — COMMENTED** [review]
Also add 'Scope' keyword

**Comment by kunalsharma-repo** [comment]
> Also add 'Scope' keyword

Added Scope keyword

File: src/test/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParserTest.java [hunk 1]
```diff
@@ -266,4 +266,10 @@ public void testSubQuery() {
       Collections.singletonList(new SObjectDescriptor("Contacts", fields)),
       result.getChildSObjects());
   }
+
+  @Test
+  public void testParseFailure() {
```
Comment (rmstar) [inline src/test/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParserTest.java]: Rename to something like testKeywordInQuery

File: src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryVisitor.java [hunk 2]
```diff
@@ -158,6 +158,30 @@ public SObjectDescriptor visitStatement(SOQLParser.StatementContext ctx) {
...
+        throw new SOQLParsingException("Invalid column name: " + ctx.getText());
+      }
+
+      if (nameParts.size() > 1) {
+        // if field contains alias, remove it to get actual field name
+        // Example:
+        // select Opportunity.Name from Opportunity -> Name
+        // select o.Name from Opportunity o -> Name
+        String topParent = nameParts.get(0);
+        if (topParent.equalsIgnoreCase(objectName) || topParent.equals(objectAlias)) {
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryVisitor.java:177]: why do we ignore case when checking equality with objectName, but not with objectAlias?

File: src/test/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParserTest.java (hunk 1 above)
Comment (kunalsharma-repo) [reply src/test/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParserTest.java]: Method renamed from testParseFailure to testKeywordInQuery

File: src/main/antlr4/soql/SOQL.g4 [hunk 3]
```diff
@@ -61,6 +61,10 @@ keyword             : GROUP
                     | CATEGORY
                     ;
 
+selectKey           : DATA
+                    | SCOPE
```
Comment (snehalakshmisha) [inline src/main/antlr4/soql/SOQL.g4:65]: We are only fixing the issue for Data and Scope keywords. Do we also want to consider other keywords like end, etc?

**Comment by vikasrathee-cs** [comment]
Closing this PR as it is having cla issue.
Raised a new PR.
https://github.com/data-integrations/salesforce/pull/131

**Pull request closed without merging**